draft-ietf-tls-esni.md

```diff
@@ -174,14 +179,32 @@ ClientHelloInner in an "encrypted_client_hello" extension, which this document
 defines ({{encrypted-client-hello}}). Finally, it sends ClientHelloOuter to the
 server.
 
-Upon receiving ClientHelloOuter, the client-facing server processes the
-encrypted ClientHelloInner and either terminates the connection (in Shared Mode)
-or forwards it to the backend server (in Split Mode).
+Upon receiving the ClientHelloOuter, the server takes one of the following
+actions:
+
+1. If the client-facing server does not support ECH, it ignores the
+   "encrypted_client_hello" extension and proceeds with the handshake as usual,
+   per {{RFC8446}}, Section 4.1.2.
+1. If the client-facing server supports ECH but cannot decrypt it, then it
+   ignores the extension and proceeds with the handshake as usual. This is
+   referred to as "ECH rejection". When ECH is rejected, the server sends an
+   acceptable ECH configuration in its EncryptedExtensions message.
+1. If the client-facing server supports ECH and can decrypt it, then it forwards
+   the CllientHelloInner to the backend, who terminates the connection. This is
+   referred to as "ECH acceptance".
+
+Upon receiving the server's response, the client must first determine whether
```

Comment:
I'm not sure this level of detail is needed here. Perhaps we can simply say, "The client determines whether ECH was accepted or rejected and proceeds accordingly. The details for each procedure are described in {{section}}."

Reply:
The high-level description is incomplete, since it doesn't talk about how the client processes the server's response. I think it's good to be complete, even if high-level.

Reply:
What I proposed was high level, I think. We should avoid redundancies in the spec where possible.

Reply:
I think of it as a gentle introduction to the protocol flow, which, IMO, aids understanding later on. I think the slight redundancy is worth it. If you don't think it is, then I'll revert this.

Reply:
Let's flush it out first and then we can decide if it's too deep? I commented on WIP text.

Reply:
Sounds good, I don't see your comment though.

Reply:
It's the first comment in this thread :)

Reply:
Applied your suggested change.